apps/web/src/modules/job/pages/index.tsx

```diff
@@ -1,15 +1,30 @@
+import { GetJobDto } from '@modela/dtos'
+import { apiClient } from 'common/utils/api'
 import { useRouter } from 'next/router'
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 
 import Card from '../components/Card'
 
 export default function JobDetail() {
   const router = useRouter()
   const { jid } = router.query
+  const [job, setJob] = useState<GetJobDto>()
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const res = (await apiClient.get('/job/' + jid)).data as GetJobDto
+        setJob(res)
+      } catch (e) {
+        console.log(e)
+        router.push('/404')
+      }
+    }
+    fetchData()
+  }, [])
```

Comment:
Why don't add jid here? Currently it call API with Nan value which push user to 404 page (Which is not even the correct error code)